app/frontend/src/features/communities/CommunityPage/JoinCommunityButton.test.tsx

```diff
@@ -0,0 +1,70 @@
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import {
+  JOIN_COMMUNITY,
+  LEAVE_COMMUNITY,
+} from "features/communities/constants";
+import { useCommunity } from "features/communities/hooks";
+import { service } from "service";
+import mockCommunity from "test/fixtures/community.json";
+import wrapper from "test/hookWrapper";
+import { assertErrorAlert, mockConsoleError, MockedService } from "test/utils";
+
+import JoinCommunityButton from "./JoinCommunityButton";
+
+let member = false;
+
+const getCommunityMock = service.communities.getCommunity as MockedService<
+  typeof service.communities.getCommunity
+>;
+const joinCommunityMock = service.communities.joinCommunity as MockedService<
+  typeof service.communities.joinCommunity
+>;
+const leaveCommunityMock = service.communities.leaveCommunity as MockedService<
+  typeof service.communities.leaveCommunity
+>;
+
+getCommunityMock.mockImplementation(async () => ({ ...mockCommunity, member }));
+joinCommunityMock.mockImplementation(async () => {
+  member = true;
+});
+leaveCommunityMock.mockImplementation(async () => {
+  member = false;
+});
+function View() {
+  const community = useCommunity(2);
+  return community.data ? (
+    <JoinCommunityButton community={community.data} />
+  ) : null;
+}
+
+describe("JoinCommunityButton", () => {
```

Comment:
These don't work. I'm not sure how to properly mock, because the button just takes the `community` object directly, but modifies it by calling the api and invalidating